pkg/lib/translation/service.go

```diff
@@ -220,9 +217,14 @@ func (s *Service) SMSMessageData(msg *MessageSpec, args interface{}) (*SMSMessag
 	}, nil
 }
 
-func (s *Service) WhatsappMessageData(language string, msg *MessageSpec, args interface{}) (*WhatsappMessageData, error) {
-	data := map[string]interface{}{}
-	template.Embed(data, args)
+func (s *Service) WhatsappMessageData(language string, msg *MessageSpec, variables *PartialTemplateVariables) (*WhatsappMessageData, error) {
+	uiParam := uiparam.GetUIParam(s.Context)
+	data := PrepareFromPartialTemplateVariables(variables, &UIParams{
+		ClientID:  uiParam.ClientID,
+		State:     uiParam.State,
+		XState:    uiParam.XState,
+		UILocales: uiParam.UILocales,
+	}, s.StaticAssets.StaticAssetURL)
```

Comment:
Please add new tests for EmailMessageData, SMSMessageData, and WhatsappMessageData to show the public API does not change.